src/pages/index.js

```diff
@@ -1,90 +1,250 @@
+      })
+      .finally(() => {
+        deletePlaceModal.setLoading(false, "Removing...", "Yes");
+      });
+  });
+  deletePlaceModal.open();
+}
+
+/* Create Cards */
+function getCardElement(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeReact
+  );
+  return card.getView();
+}
+
+/* GET Cards */
+let section;
+api
+  .getInitialCards()
+  .then((items) => {
+    section = new Section(
+      {
+        items: items,
+        renderer: (cardData) => {
+          const cardElement = getCardElement(cardData);
+          return cardElement;
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
+  })
+  .catch((err) => console.error(err));
+
+/* user info instance */
+const userInfo = new UserInfo({
+  profileName: ".profile__title",
+  profileJob: ".profile__description",
+  profileAvatar: ".profile__image",
+});
+
+/* GET User Info */
+api
+  .getUserInfo()
+  .then((profileData) => {
+    userInfo.setUserInfo({
+      name: profileData.name,
+      about: profileData.about,
+    });
+    userInfo.setUserAvatar({ avatar: profileData.avatar });
+  })
+  .catch((err) => console.error(err));
+
+/* PATCH Profile Edit Function */
+function handleProfileEditSubmit(inputValues) {
+  editProfileModal.setLoading(true);
+
+  if (!inputValues.name || !inputValues.about) {
+    console.error("Please fill in all the fields");
+    return;
+  }
+
+  api
+    .editProfile({
+      name: inputValues.name,
+      about: inputValues.about,
+    })
+    .then(({ name, about }) => {
+      userInfo.setUserInfo({
+        name: name,
+        about: about,
+      });
+      editProfileModal.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user info:", err);
+    })
+    .finally(() => {
+      editProfileModal.setLoading(false);
+    });
+}
+
+/* PATCH Profile Avatar Function */
+function handleAvatarSubmit(inputValues) {
+  updateAvatarModal.setLoading(true);
+
+  api
+    .updateAvatar(inputValues)
+    .then((res) => {
+      userInfo.setUserAvatar(res);
+      updateAvatarModal.reset();
+      updateAvatarModal.close();
+      console.log("Success:", res);
+    })
+    .catch((err) => {
+      console.error("Error updating profile avatar:", err);
+    })
+    .finally(() => {
+      updateAvatarModal.setLoading(true, "Save");
+    });
+}
+
+/* POST Add Place Function */
+function handleNewPlaceSubmit(inputValues) {
+  addPlaceModal.setLoading(true);
+
+  api
+    .addCard({ name: inputValues.name, link: inputValues.link })
+    .then(({ name, link, _id }) => {
+      section.addItem(getCardElement({ name, link, _id }));
+      addPlaceModal.close();
+
+      addPlaceModal.reset();
+    })
+    .catch((err) => {
+      console.error("Error adding new place:", err);
+    })
+    .finally(() => {
+      addPlaceModal.setLoading(false);
+      addPlaceValidation.disableButton();
```

Comment:
Needs correction: there is no need to disable the button in the case of an error